v6.5.0 (#9036)
mui/mui-x: CHANGELOG.md

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## v6.5.0

_May 19, 2023_

We'd like to offer a big thanks to the 10 contributors who made this release possible. Here are some highlights ✨:

> - 💫 Introduce filtering on column headers for `DataGridPro`:

Reviewer: ```suggestion
- 💫 Introduce filtering on column headers for `DataGridPro` and `DataGridPremium`:
```

Or is it totally obvious ?

Reviewer 2: `- 💫 Introduce filtering on column headers for commercial users` 

could also be an option 🤷‍♂️

  <!-- TODO: Add a nice demo -->

>   [...]

Author: @MBilalShafi can we do a nice-looking demo to brag more 😁. Also, let me know if you want to extend the highlight with more info.

>   <!-- TODO: Add a nice demo -->
>   [...]

Reviewer 2: Not sure if it's nice looking but here's a try 😄 
```suggestion
  https://github.com/mui/mui-x/assets/12609561/c4c2bfec-59cf-4cab-932d-dc1983081de9
```

Reviewer 3: Nice video!

  See [the documentation](https://mui.com/x/react-data-grid/filtering/#header-filters) for more information

- 🌍 Improve Hebrew (he-IL) and Czech (cs-CZ) locales

> - ⚙️ Performance improvements

Reviewer 2: ```suggestion
- 📝 Support for editing on pinned rows
- 🚀 Performance improvements
```

- 🐞 Bugfixes
- 📚 Documentation improvements

### `@mui/x-data-grid@6.5.0` / `@mui/x-data-grid-pro@6.5.0` / `@mui/x-data-grid-premium@6.5.0`

#### Changes

- [DataGrid] Fix grid size calculation when `.MuiDataGrid-main` has border (#8882) @cherniavskii
- [DataGridPro] Filtering on Column Header (#7760) @MBilalShafi
- [DataGridPro] Improve `treeData` and `rowGrouping` performance (#8990) @MBilalShafi
- [DataGridPro] Support pinned rows editing (#8921) @cherniavskii

> - [l10n] Add missing keys in Hebrew (he-IL) locale (#8943) @Itzik-Tech

Reviewer 4: ```suggestion
- [l10n] Improve Hebrew (he-IL) locale (#8943) @Itzik-Tech
```

- [l10n] Improve Czech (cs-CZ) locale (#8829) @harastaivan

> - [l10n] Update Czech (cs-CZ) locale (#8956) @davidzemancz

Reviewer 4: ```suggestion
- [l10n] Improve Czech (cs-CZ) locale (#8956) @davidzemancz
```

### `@mui/x-date-pickers@6.5.0` / `@mui/x-date-pickers-pro@6.5.0`

#### Changes

> - [fields] Fix selection of first element on DateField (#9005) @noraleonte

Reviewer 4: ```suggestion
- [fields] Select the first section instead of last when clicking right of content (#9005) @noraleonte
```

> - [fields] Refacto prop drilling in fields (#8660) @flaviendelangle

Reviewer 4: ```suggestion
- [fields] Refactor prop drilling in fields (#8660) @flaviendelangle
```

- [pickers] Allow to render the months before `currentMonth` instead of the one after (#8592) @flaviendelangle

> - [pickers] Fix view management when `openTo` or `views` is modified (#8997) @alexfauquette

Reviewer 4: ```suggestion
- [pickers] Fix view management when `openTo` or `views` is modified (#8997) @alexfauquette
- [l10n] Improve Czech (cs-CZ) locale (#8829) @harastaivan
```

### Docs

- [docs] Clarify what Controlled / Uncontrolled means (#8926) @flaviendelangle
- [docs] Fix docs using wrong service worker (#9030) @cherniavskii
- [docs] Remove prop-types from JS demos (#9008) @flaviendelangle

### Core

- [core] Add assertion about checkbox rerenders (#8974) @oliviertassinari
- [core] Fix `yarn.lock` (#8988) @flaviendelangle
- [core] Fix flacky adapter test (#8995) @flaviendelangle
- [charts] Clean the axis rendering (#8948) @alexfauquette

> - [Datagrid] Memoize root props (#8942) @romgrk

Reviewer 2: Also move to DataGrid changes?
```suggestion
- [DataGrid] Memoize root props for better performance (#8942) @romgrk
```

- [DataGridPro] Filtering on Column Header (#7760) @MBilalShafi
- [DataGridPro] Improve `treeData` and `rowGrouping` performance (#8990) @MBilalShafi
- [DataGridPro] Support pinned rows editing (#8921) @cherniavskii
- [l10n] Improve Hebrew (he-IL) locale (#8943) @Itzik-Tech
- [l10n] Improve Czech (cs-CZ) locale (#8829) @harastaivan
- [l10n] Improve Czech (cs-CZ) locale (#8956) @davidzemancz

### `@mui/x-date-pickers@6.5.0` / `@mui/x-date-pickers-pro@6.5.0`

#### Changes

- [fields] Select the first section instead of last when clicking right of content (#9005) @noraleonte
- [fields] Refactor prop drilling in fields (#8660) @flaviendelangle
- [pickers] Allow to render the months before `currentMonth` instead of the one after (#8592) @flaviendelangle
- [pickers] Fix view management when `openTo` or `views` is modified (#8997) @alexfauquette
- [l10n] Improve Czech (cs-CZ) locale (#8829) @harastaivan

### Docs

- [docs] Clarify what Controlled / Uncontrolled means (#8926) @flaviendelangle
- [docs] Fix docs using wrong service worker (#9030) @cherniavskii
- [docs] Remove prop-types from JS demos (#9008) @flaviendelangle

### Core

> - [core] Add assertion about checkbox rerenders (#8974) @oliviertassinari

Reviewer 4: ```suggestion
- [core] Add assertion about checkbox rerenders (#8974) @oliviertassinari
- [core] Allow selecting a section by type in field tests (#9009) @flaviendelangle
```
